src/persist-client/src/impl/machine.rs

```diff
@@ -409,23 +409,42 @@ where
         cmd: &CmdMetrics,
         mut work_fn: WorkFn,
     ) -> Result<(SeqNo, Result<R, E>, RoutineMaintenance), Indeterminate> {
+        let is_write = cmd.name == self.metrics.cmds.compare_and_append.name;
         cmd.run_cmd(|cas_mismatch_metric| async move {
             let path = self.shard_id().to_string();
+            let mut garbage_collection;
 
             loop {
-                let (work_ret, new_state) = match self.state.clone_apply(&mut work_fn) {
+                let (work_ret, mut new_state) = match self.state.clone_apply(&mut work_fn) {
                     Continue(x) => x,
                     Break(err) => {
                         return Ok((self.state.seqno(), Err(err), RoutineMaintenance::default()))
                     }
                 };
+
+                // If this command has been selected to perform gc, fire off a
```

Comment:
maybe clarify that `If this command has been selected to perform gc` means to perform `compare_and_set` for this seqno (&& there even if gc work to perform)

Reply:
done!